granulate_utils/metadata/databricks_client.py

```diff
@@ -135,12 +129,54 @@ def _cluster_all_tags_metadata(self) -> Optional[Dict[str, str]]:
-                try:
-                    all_tags_value = json.loads(prop[1])
-                except Exception as e:
-                    raise DatabricksJobNameDiscoverException(f"Failed to parse {prop=}") from e
-                return {cluster_all_tag["key"]: cluster_all_tag["value"] for cluster_all_tag in all_tags_value}
+        # Creating a dict of the relevant properties and their values.
+        service_name_prop_candidates = {
+            prop[0]: prop[1] for prop in props if (CLUSTER_ALL_TAGS_PROP == prop[0] or CLUSTER_NAME_PROP == prop[0])
+        }
+        if len(service_name_prop_candidates) == 0:
+            # We expect at least one of the properties to be present.
+            raise DatabricksJobNameDiscoverException(f"Failed to create dict of relevant properties {env=}")
+        # First, trying to extract `CLUSTER_TAGS_KEY` property, in case not redacted.
+        if (
+            cluster_all_tags_value := service_name_prop_candidates.get(CLUSTER_ALL_TAGS_PROP)
+        ) is not None and "redacted" not in cluster_all_tags_value:
+            try:
+                cluster_all_tags_value_json = json.loads(cluster_all_tags_value)
+            except Exception as e:
+                raise DatabricksJobNameDiscoverException(f"Failed to parse {cluster_all_tags_value}") from e
+            return self._apply_pattern(
+                {cluster_all_tag["key"]: cluster_all_tag["value"] for cluster_all_tag in cluster_all_tags_value_json}
+            )
+        # As a fallback, trying to extract `CLUSTER_NAME_PROP` property.
+        elif (cluster_name_value := service_name_prop_candidates.get(CLUSTER_NAME_PROP)) is not None:
+            return self._apply_pattern({CLUSTER_NAME_KEY: cluster_name_value})
         else:
-            raise DatabricksJobNameDiscoverException(f"Failed to find {CLUSTER_TAGS_KEY=} in {props=}")
+            raise DatabricksJobNameDiscoverException(
+                f"Failed to extract {CLUSTER_ALL_TAGS_PROP} or {CLUSTER_NAME_PROP} from {props=}"
+            )
+
+    @staticmethod
+    def _apply_pattern(metadata: Dict[str, str]) -> Dict[str, str]:
+        """
+        Applies certain patterns on the metadata values.
+        We mostly use the metadata values as service names, so we want to make sure the metadata values
+        match some service name requirements.
+
+        e.g.: Job Name might include spaces, we want to replace them with dashes.
+        """
+        if JOB_NAME_KEY in metadata:
+            metadata[JOB_NAME_KEY] = metadata[JOB_NAME_KEY].replace(" ", "-").lower()
+        if CLUSTER_NAME_KEY in metadata:
+            # We've tackled cases where the cluster name includes Run ID, we want to remove it.
+            metadata[CLUSTER_NAME_KEY] = re.sub(RUN_ID_REGEX, "", metadata[CLUSTER_NAME_KEY])
+            metadata[CLUSTER_NAME_KEY] = metadata[CLUSTER_NAME_KEY].replace(" ", "-").lower()
+        return metadata
+
+    @property
+    def get_name_from_metadata(self) -> Optional[str]:
```

Comment:
I prefer if the property itself won't have a `get_` inside it. Maybe something like this:
```suggestion
    def job_name(self) -> Optional[str]:
```

Reply:
3f8726c